samples/test/secretmanager.test.js

```diff
@@ -0,0 +1,151 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+'use strict';
+
+const { assert } = require('chai');
+const cp = require('child_process');
+const uuidv4 = require('uuid/v4');
+
+const { SecretManagerServiceClient } = require('@google-cloud/secret-manager');
+const client = new SecretManagerServiceClient();
+
+const projectId = process.env.GCLOUD_PROJECT;
+const secretId = uuidv4();
+const payload = `my super secret data`;
+
+let secret;
+let version;
+
+const execSync = cmd => cp.execSync(cmd, { encoding: 'utf-8' });
+
+describe(`Secret Manager samples`, () => {
+  before(async () => {
+    [secret] = await client.createSecret({
+      parent: `projects/${projectId}`,
+      secretId: secretId,
+      secret: {
+        replication: {
+          automatic: {},
+        },
+      },
+    });
+
+    [version] = await client.addSecretVersion({
+      parent: secret.name,
+      payload: {
+        data: Buffer.from(payload),
+      },
+    });
+  });
+
+  after(async () => {
+    try {
+      await client.deleteSecret({
+        name: secret.name,
+      });
+    } catch (e) {
+      // Ignore errors
```

Comment:
It might be better to fail tests if we're in a position where we create a secret and can't delete it? I'm hoping this almost never happens (I don't want us to create a ton of garbage data accidentally with a bad cleanup step).

Reply:
I made it only catch NOT_FOUND errors (because that means the cleanup already happened in a step).